### PR TITLE
http3: make error codes public and consistent with http2 package

### PR DESCRIPTION
**Pull request by jfgiorgi**
simple renaming and making public error code values. 
consistent with https://github.com/golang/net/blob/master/http2/errors.go
partial fix for #3737

### CODE REVIEW

**Comment by marten-seemann** [comment]
Does this allow you to get the context cancelation error?

**Comment by jfgiorgi** [comment]
> Does this allow you to get the context cancelation error?

not it's just a renaming of exiting private values. It just  allows us to use `http3.ErrRequestCanceled` instead of 0x10c which is already a good thing.

Context cancellation is more complex to do (unless we treat all `ErrRequestCanceled` as a context cancellation which we probably should not).

File: http3/error_codes.go [hunk 1]
```diff
@@ -6,66 +6,66 @@ import (
...
-	errorSettingsError        errorCode = 0x109
-	errorMissingSettings      errorCode = 0x10a
-	errorRequestRejected      errorCode = 0x10b
-	errorRequestCanceled      errorCode = 0x10c
-	errorRequestIncomplete    errorCode = 0x10d
-	errorMessageError         errorCode = 0x10e
-	errorConnectError         errorCode = 0x10f
-	errorVersionFallback      errorCode = 0x110
-	errorDatagramError        errorCode = 0x4a1268
+	ErrrNoError             ErrCode = 0x100
```
Comment (marten-seemann) [inline http3/error_codes.go:12]: Typo.

Comment (kgersen) [reply http3/error_codes.go]: typo fixed

File: http3/error_codes.go [hunk 2]
```diff
@@ -6,66 +6,66 @@ import (
...
-	errorMissingSettings      errorCode = 0x10a
-	errorRequestRejected      errorCode = 0x10b
-	errorRequestCanceled      errorCode = 0x10c
-	errorRequestIncomplete    errorCode = 0x10d
-	errorMessageError         errorCode = 0x10e
-	errorConnectError         errorCode = 0x10f
-	errorVersionFallback      errorCode = 0x110
-	errorDatagramError        errorCode = 0x4a1268
+	ErrNoError              ErrCode = 0x100
+	ErrGeneralProtocolError ErrCode = 0x101
```
Comment (marten-seemann) [inline http3/error_codes.go:13]: The H2 package uses slightly different names, adding `Code` to the name (e.g. `ErrCodeGeneralProtocolError`). Should we do the same?

**Review by marten-seemann — COMMENTED** [review]
Sorry for the late review. This looks good to me. I'd still like to actually resolve the issue, but this is a well-contained change, so let's just get this PR merged.

One question regarding naming consistency with the standard library, see below.

File: http3/error_codes.go (hunk 2 above)
Comment (kgersen) [reply http3/error_codes.go:13]: I think (but not I've not verified) the Go team convention/idiomatic naming if that the named values of a pseudo "enum"  have to all start with the full name of the type or not. So ErrCode*. It's a good convention when using completion in an IDE for instance.

Comment (marten-seemann) [reply http3/error_codes.go:13]: Can you make the change then?

**Review by marten-seemann — APPROVED** [review]

**Comment by marten-seemann** [comment]
Thank you @kgersen!

**Pull request merged**